# **FINAL REPORT**

<span style="font-size: 15px;">
Broad question: What player characteristics and behaviours are most predctive of subscribing to a game-related newsletter, and how do these features differ across various player types?

Specific question: Can the amount of hours playing video games and the age of the player predict their subscription status to a gaming newsletter? Additionally, does this predictive relationship differ across players of different expereince levels?
</span>

### Background Information

<span style="font-size: 15px;">

A reaearch group in Computer Science at UBC collected data about how people play video games. They are recording players' actions actions as the players navigate through the Minecraft server that the group have set up. They need to hone in on specific recruitment efforts, and make sure they have enough resources (i.e., software licesnses) to handle the number of players they attract. Through our final report, we are targetting the effects that the variables played_hours and age have on the players' subscription status.

<span>


In [3]:
library(tidyverse)
url <- ("https://raw.githubusercontent.com/msyr125/DSCI-100-GROUP-PROJECT/refs/heads/main/players.csv")
players <- read_csv(url)

head(players)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

### Data Description

<span style="font-size: 15px;">
The dataset players.csv contains information about individual Minecraft players collected by UBC research group studying player behaviour. It includes demogrpahoc data, playtime, and expereince level for each player along with whether thy subscribed to a game-related newsletter. 

**Number of observations:** 196 <br>
**Number of variables:** 7 <br>
**File name:** players.csv
</span>

#### **Summary Statistics**
<span style="font-size: 15px;">
    
**Experience:**
* Pro = 14
* Veteran = 48
* Regular = 36
* Amateur = 63
* Beginner = 35

**Subscribers:**
* Number of people subscribed = 144
* Number of people not subscribed = 52

**Hours Played:**
* Minimum hours played = 0
* Maximum hours played = 223.1
* Mean hours played = 5.85
* Standard deviation of hours played = 28.36

**Age of Players:**
* Minimum age of player = 9
* Maximum age of player = 58
* Mean age of player = 21.14
* Standard deviation of player age = 7.39

**Gender Distribution of Players:**
* Male players = 124
* Female players = 37
* Non-binary players = 7
* Two-Spirited players = 6
* Prefer not to say = 11
</span>

In [6]:
# Summary Statistics performed

filter_pro <- players |>
    filter(experience == 'Pro')
## Pro = 14

filter_amateur <- players |>
    filter(experience == 'Amateur')
## Amateur = 63

filter_vet <- players |>
    filter(experience == 'Veteran')
## Veteran = 48

filter_beg <- players |>
    filter(experience == 'Beginner')
## Beginner = 35

filter_reg <- players |>
    filter(experience == 'Regular')
## Regular = 36

filter_sub <- players |>
    filter(subscribe == 'FALSE')
## TRUE = 144, FALSE = 52

min_hrs <- players |>
    pull(played_hours) |>
    min(na.rm = TRUE)
## Min hrs = 0

max_hrs <- players |>
    pull(played_hours) |>
    max(na.rm = TRUE)
## Max hrs = 223.1

mean_hrs <- players |>
    pull(played_hours) |>
    mean(na.rm = TRUE)
## Mean = 5.85

stdev_hrs <- players |>
    pull(played_hours) |>
    sd(na.rm = TRUE)
## Standard deviation = 28.36

min_age <- players |>
    pull(Age) |>
    min(na.rm = TRUE)
## Min age = 9

max_age <- players |>
    pull(Age) |>
    max(na.rm = TRUE)
## Max age = 58

mean_age <- players |>
    pull(Age) |>
    mean(na.rm = TRUE)
## Age = 21.14

stdev_age <- players |>
    pull(Age) |>
    sd(na.rm = TRUE)
## Standard deviation = 7.39

male <- players |>
    filter(gender == 'Male')
## Male players = 124

female <- players |>
    filter(gender == 'Female')
## Female players = 37

non_binary <- players |>
    filter(gender == 'Non-binary')
## Non-binary players =7

two_spirited <- players |>
    filter(gender == 'Two-Spirited')
## Two-Spirited players = 6

none <- players |>
    filter(gender == 'Prefer not to say')
## Prefer not to say = 11